File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/repository/redis/RedisRepository.java
package org.javamaster.invocationlab.admin.service.repository.redis;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Set;

/**
 * 封装redis相关的操作
 *
 * @author yudong
 */
@SuppressWarnings("unchecked")
@Repository
public class RedisRepository implements org.javamaster.invocationlab.admin.service.repository.Repository {

    @Autowired
    private RedisTemplate<Object, Object> redisTemplate;

    public void setAdd(String key, Object value) {
        redisTemplate.opsForSet().add(key, value);
    }

    public <T> Set<T> members(String key) {
        return (Set<T>) redisTemplate.opsForSet().members(key);
    }

    public void setRemove(String key, Object value) {
        redisTemplate.opsForSet().remove(key, value);
    }


    public void mapPut(String key, Object hashKey, Object value) {
        redisTemplate.opsForHash().put(key, hashKey, value);
        redisTemplate.persist(key);
    }

    @SuppressWarnings("unchecked")
    public <T> T mapGet(String key, Object hashKey) {
        return (T) redisTemplate.opsForHash().get(key, hashKey);
    }

    public <T> Set<T> mapGetKeys(String key) {
        return (Set<T>) redisTemplate.opsForHash().keys(key);
    }

    public Boolean delete(String key) {
        return redisTemplate.delete(key);
    }

    public <T> List<T> mapGetValues(String key) {
        return (List<T>) redisTemplate.opsForHash().values(key);
    }

    public void removeMap(String key, String hashKey) {
        redisTemplate.opsForHash().delete(key, hashKey);
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/repository/redis/RedisRepository.kt
package org.javamaster.invocationlab.admin.service.repository.redis

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Repository

/**
 * 封装redis相关的操作
 *
 * @author yudong
 */
@Suppress("UNCHECKED_CAST")
@Repository
class RedisRepository : org.javamaster.invocationlab.admin.service.repository.Repository {
    @Autowired
    private lateinit var redisTemplate: RedisTemplate<Any, Any>

    fun setAdd(key: String, value: Any) {
        redisTemplate.opsForSet().add(key, value)
    }

    fun <T> members(key: String): Set<T>? {
        return redisTemplate.opsForSet().members(key) as Set<T>
    }

    fun setRemove(key: String, value: Any?) {
        redisTemplate.opsForSet().remove(key, value)
    }


    fun mapPut(key: String, hashKey: Any, value: Any) {
        redisTemplate.opsForHash<Any, Any>().put(key, hashKey, value)
        redisTemplate.persist(key)
    }

    fun <T> mapGet(key: String, hashKey: Any): T? {
        return redisTemplate.opsForHash<Any, Any>()[key, hashKey] as T
    }

    fun <T> mapGetKeys(key: String): Set<T> {
        return redisTemplate.opsForHash<Any, Any>().keys(key) as Set<T>
    }

    fun delete(key: String): Boolean {
        return redisTemplate.delete(key)
    }

    fun <T> mapGetValues(key: String): List<T> {
        return redisTemplate.opsForHash<Any, Any>().values(key) as List<T>
    }

    fun removeMap(key: String, hashKey: String?) {
        redisTemplate.opsForHash<Any, Any>().delete(key, hashKey)
    }
}
